server.c

```diff
@@ -2813,6 +2847,9 @@ server_main(struct nsd *nsd)
 	unlink(nsd->zonestatfname[0]);
 	unlink(nsd->zonestatfname[1]);
 #endif
+#ifdef BIND8_STATS
+	unlink(nsd->statfname);
```

Comment:
Maybe we should add a `server_stat_free` to handle cleanup? The inverse of `server_stat_alloc`, may make reasoning about these operations easier.

Reply:
The `server_stat_free` function is added in this commit https://github.com/NLnetLabs/nsd/pull/292/commits/9b4090bc912fd9933bf6480338c9c93ee49d8dad . Maybe that is easier to deal with.